core/trino-main/src/main/java/io/trino/sql/analyzer/TypeSignatureProvider.java

```diff
@@ -90,7 +90,7 @@ public TypeSignature getTypeSignature(List<Type> boundTypeParameters)
     public String toString()
     {
         if (hasDependency) {
-            return super.toString();
+            return "<lambda>";
```

Comment:
Technically, this should be `<function>`, as that is the type of the argument. Lambda expressions are one way to create functions, but not the only one (e.g., bind, and in the future, maybe function-returning functions).